staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -131,6 +131,9 @@ type ClientConfigLoadingRules struct {
 	// WarnIfAllMissing indicates whether the configuration files pointed by KUBECONFIG environment variable are present or not.
 	// In case of missing files, it warns the user about the missing files.
 	WarnIfAllMissing bool
+
+	// WarningHandler is the warning log callback to use in case of missing files.
+	WarningHandler func(string)
```

Comment:
I'd like this being set in https://github.com/kubernetes/kubernetes/blob/f6152d152140f42815765c7d9703b9817943c345/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go#L132 which will ensure this is being used and actually consumed by kubectl itself.